test/main.sh

```diff
@@ -616,6 +616,12 @@ run_test test_snap_restore "snapshot restores"
 run_test test_config_profiles "profiles and configuration"
 run_test test_config_edit "container configuration edit"
 run_test test_config_edit_container_snapshot_pool_config "container and snapshot volume configuration edit"
+run_test test_metadata_show "container metadata show"
+run_test test_metadata_edit "contaimer metadata edit"
+run_test test_metadata_template_list "contaimer template list"
+run_test test_metadata_template_content "contaimer template content"
+run_test test_metadata_template_add "container template add"
+run_test test_metadata_template_remove "container template remove"
```

Comment:
https://jenkins.linuxcontainers.org/view/LXD/view/LXD%20builds/job/lxd-github-pull-test/2763/arch=amd64,backend=dir,compiler=golang-1.6,restrict=lxd-priv/consoleFull will show you the problem. The list of sections on the left already takes several seconds to get extracted and rendered. Gets worse every time we add more stuff.

I agree that "basic_usage" includes way too much stuff including a bunch of duplication with other tests.

What I'd like to see is one "run_test" per feature, so a "test_container_metadata" one is fine, but having things split further just makes things harder, both for dealing with jenkins and if manually running tests (as there's no way I'll remember test_metadata_template_remove so will have to always look at test/main.sh...).